Description: configure for Debian system
 Set soversion and add pkg-config support.
 .
 libshadowdive (0.1.0-1) unstable; urgency=low
 .
   * Initial release of debianized sources.
Author: Michele Bini <michele.bini@gmail.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- libshadowdive-0.1.0.orig/CMakeLists.txt
+++ libshadowdive-0.1.0/CMakeLists.txt
@@ -1,6 +1,9 @@
 cmake_minimum_required(VERSION 2.6)
 project(ShadowDive C)
 
+set(SHADOWDIVE_MAJOR_VERSION 0)
+set(SHADOWDIVE_VERSION 0.1.0)
+
 SET(CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake-scripts)
 
 set(CMAKE_C_FLAGS "-Wall -std=c99 -ggdb -pedantic")
@@ -13,7 +16,7 @@ include_directories(include/)
 enable_testing()
 add_subdirectory(testing)
 
-add_library(shadowdive
+add_library(shadowdive SHARED
     src/af.c
     src/animation.c
     src/bk.c
@@ -34,6 +37,11 @@ add_library(shadowdive
     src/internal/helpers.c
 )
 
+set_target_properties(shadowdive PROPERTIES
+  VERSION ${SHADOWDIVE_VERSION}
+  SOVERSION ${SHADOWDIVE_MAJOR_VERSION}
+  )
+
 add_executable(test_main test/test_main.c)
 add_executable(test_sounds test/sounds_main.c)
 add_executable(test_roundtrip test/roundtrip.c)
@@ -48,8 +56,14 @@ target_link_libraries(test_font shadowdi
 target_link_libraries(test_stringparser shadowdive)
 target_link_libraries(test_language shadowdive)
 
+configure_file(libshadowdive.pc.in libshadowdive.pc @ONLY)
+install(
+  FILES ${CMAKE_CURRENT_BINARY_DIR}/libshadowdive.pc
+  DESTINATION lib/pkgconfig
+  )
+
 FILE(GLOB H_FILES "include/shadowdive/*.h")
-INSTALL(FILES ${H_FILES} DESTINATION include/shadowdive/)
+INSTALL(FILES ${H_FILES} DESTINATION include/shadowdive)
 INSTALL(TARGETS shadowdive
     RUNTIME DESTINATION bin
     LIBRARY DESTINATION lib
--- /dev/null
+++ libshadowdive-0.1.0/libshadowdive.pc.in
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${exec_prefix}/lib
+includedir=${prefix}/include/shadowdive
+
+Name: shadowdive
+Description: Parsing and writing One Must Fall 2097 file formats.
+Version: @SHADOWDIVE_VERSION@
+Libs: -L${libdir}
+Cflags: -I${includedir}
